src/main/java/net/dv8tion/jda/api/entities/User.java

```diff
@@ -225,4 +228,168 @@ default String getEffectiveAvatarUrl()
      */
     @Nonnull
     JDA getJDA();
+
+    /**
+     * Returns the {@link net.dv8tion.jda.api.entities.User.Flag Flags} of this user.
+     * 
+     * @return EnumSet containing the flags of the user.
+     */
+    @Nonnull
+    EnumSet<Flag> getFlags();
+
+    /**
+     * Represents the bit offsets used by Discord for public flags
+     */
+    enum Flag
+    {
+        STAFF(                     0, "Discord Employee"),
+        PARTNER(                   1, "Discord Partner"),
+        HYPESQUAD(                 2, "HypeSquad Events"),
+        BUG_HUNTER_LEVEL_1(        3, "Bug Hunter Level 1"),
+
+        // HypeSquad
+        HYPESQUAD_BRAVERY(         6, "HypeSquad Bravery"),
+        HYPESQUAD_BRILLIANCE(      7, "HypeSquad Brilliance"),
+        HYPESQUAD_BALANCE(         8, "HypeSquad Balance"),
+
+        EARLY_SUPPORTER(           9, "Early Supporter"),
+        TEAM_USER(                10, "Team User"),
+        SYSTEM(                   12, "System User"),
+        BUG_HUNTER_LEVEL_2(       14, "Bug Hunter Level 2"),
+        VERIFIED_BOT(             16, "Verified Bot"),
+        VERIFIED_DEVELOPER(       17, "Verified Bot Developer"),
+        
+        UNKNOWN(                   -1, "Unknown");
+
+        /**
+         * Empty array of Flag enum, useful for optimized use in {@link Collection#toArray(Object[])}.
+         */
+        public static final Flag[] EMPTY_FLAGS = new Flag[0];
+        
+        private final int offset;
+        private final long raw;
+        private final String name;
+
+        Flag(int offset, @Nonnull String name)
+        {
+            this.offset = offset;
+            this.raw = 1 << offset;
+            this.name = name;
+        }
+
+        /**
+         * The readable name as used in the Discord Client.
+         * 
+         * @return The readable name of this {@link Flag Flag}.
+         */
+        @Nonnull
+        public String getName()
+        {
+            return this.name;
+        }
+
+        /**
+         * The binary offset of the flag.
+         * 
+         * @return The offset that represents this {@link Flag Flag}.
```

Comment:
The [JDA Structure Guide](https://github.com/DV8FromTheWorld/JDA/wiki/6)-JDA-Structure-Guide) advises against documenting with imported names. I'm not sure if it doesn't apply here for some reason, but I'm mentioning it just in case

Reply:
I blame IntelliJ for auto-cutting the {@link} parameter as it had the full path before.

Reply:
Fair enough